R/normalization.R

```diff
@@ -0,0 +1,297 @@
+#' Functions and utilities for adjustment of LC-MS/metabolomics-specific biases
+#'
+#' @noRd
+NULL
+
+#' @title Linear model-based normalization of abundance matrices
+#'
+#' @description
+#'
+#' The `fit_lm` and `adjust_lm` functions facilitate linear model-based
+#' normalization of abundance matrices. The expected noise in a numeric
+#' data matrix can be modeled with a linear regression model using the
+#' `fit_lm` function and the data can subsequently be adjusted using the
+#' `adjust_lm` function (i.e., the modeled noise will be removed from the
+#' abundance values). A typical use case would be to remove injection index
+#' dependent signal drifts in a LC-MS derived metabolomics data:
+#' a linear model of the form `y ~ injection_index` could be used to model
+#' the measured abundances of each feature (each row in a data matrix) as a
+#' function of the injection index in which a specific sample was measured
+#' during the LC-MS measurement run. The fitted linear regression models
+#' can subsequently be used to adjust the abundance matrix by removing
+#' these dependencies from the data. This allows to perform signal
+#' adjustments as described in (Wehrens et al. 2016).
+#'
+#' The two functions are described in more details below:
+#'
+#' `fit_lm` allows to fit a linear regression model (defined with parameter
+#' `formula`) to each row of the numeric data matrix submitted with parameter
+#' `y`. Additional covariates of the linear model defined in `formula` are
+#' expected to be provided as columns in a `data.frame` provided with
+#' parameter `data`.
```

Comment:
"...should be included as columns in a `data.frame` supplied via the `data` parameter." maybe ? to prevent repetition of "provided".

Reply:
fixed.